spotlight-client/src/ShareModal/ShareModal.tsx

```diff
@@ -0,0 +1,232 @@
+  cursor: pointer;
+  display: flex;
+  font-size: ${rem(11)};
+  justify-content: center;
+  padding: ${rem(12)} ${rem(16)};
+  transition: background-color ${animation.defaultDuration}ms;
+
+  &:disabled {
+    cursor: not-allowed;
+  }
+
+  &:hover,
+  &:focus {
+    background-color: ${colors.text};
+  }
+
+  &:active {
+    background-color: ${colors.text};
+  }
+`;
+
+const ShareActions = styled.div`
+  border-color: ${colors.rule};
+  border-style: solid;
+  border-width: 1px 0;
+  margin: ${rem(24)} 0;
+  padding: ${rem(24)} 0;
+  position: relative;
+`;
+
+const UrlText = styled.div`
+  font-family: ${typefaces.display};
+  font-size: ${rem(21)};
+  letter-spacing: -0.04em;
+  padding-bottom: ${rem(16)};
+`;
+
+const ShareButtons = styled.div`
+  display: flex;
+
+  & > button {
+    margin: 0 ${rem(4)};
+
+    &:first-of-type {
+      margin-left: 0;
+    }
+
+    &:last-of-type {
+      margin-right: 0;
+    }
+  }
+`;
+
+const IconTwitter = styled.img.attrs({
+  alt: "Twitter",
+  src: twitterPath,
+})``;
+
+const IconEmail = styled.img.attrs({
+  alt: "E-mail",
+  src: emailPath,
+})``;
+
+const IconFacebook = styled.img.attrs({
+  alt: "Facebook",
+  src: facebookPath,
+})``;
+
+const CopyConfirmation = styled(animated.div).attrs({ role: "status" })`
+  bottom: ${rem(4)};
+  font-size: ${rem(11)};
+  position: absolute;
+`;
+
+const IncludeSectionLabel = styled.label`
+  align-items: baseline;
+  display: flex;
+  font-size: ${rem(14)};
+  line-height: 1.4;
+`;
+
+const SectionTitle = styled.span`
+  color: ${colors.link};
+`;
+
+const ShareModal = (
+  modalProps: Omit<SpotlightModalProps, "children">
+): JSX.Element => {
+  const {
+    tenantStore: {
+      currentTenantId,
+      currentNarrative,
+      currentNarrativeTypeId,
+      currentSectionNumber,
+    },
+  } = useDataStore();
+  const [includeSection, setIncludeSection] = useState(false);
+
+  let urlToShare = window.location.host;
+
+  if (currentTenantId) {
+    if (currentNarrativeTypeId) {
+      urlToShare += getUrlForResource({
+        page: "narrative",
+        params: {
+          tenantId: currentTenantId,
+          narrativeTypeId: currentNarrativeTypeId,
+          sectionNumber: includeSection ? currentSectionNumber : undefined,
+        },
+      });
+    } else {
+      urlToShare += getUrlForResource({
+        page: "tenant",
+        params: { tenantId: currentTenantId },
+      });
+    }
+  }
+
+  const [isCopied, copyUrl] = useClipboard(urlToShare, {
+    successDuration: 5000,
+  });
+
+  const transitionConfirmation = useTransition(isCopied, null, {
+    from: { opacity: 0 },
+    enter: { opacity: 1 },
+    leave: { opacity: 0 },
+  });
+
+  return (
+    <Modal {...modalProps}>
+      <ModalHeading>Share</ModalHeading>
+      <ShareActions>
+        <UrlText>{urlToShare}</UrlText>
+        <ShareButtons>
+          <Button onClick={() => copyUrl()}>Copy URL</Button>
+          <EmailShareButton url={urlToShare}>
```

Comment:
Is there a setting to have this open in a new tab? After sending the email it closes the tab so I lose the dashboard completely.

Reply:
unfortunately there is not, but I guess it would be pretty trivial to just write my own `mailto` link here instead ... I was just using this because it was already there but that's a good point!

Reply:
after digging around some more I found that there actually is a way to do this via the library, and it actually seems to work more consistently than a plain `mailto` link with `target="_blank"` despite being slightly kludgy, so that's what I went with!